### PR TITLE
Multiple MathML blocks

### PR DESCRIPTION
**Pull request by kerimoyle**
Tests to demonstrate #556.  There are three tests here which will fail at the moment, but should be allowed: more than one MathML block in:
- a component
- a reset test_value
- a reset reset_value

Currently all return libXML2 errors about being unable to find a root node.

### CODE REVIEW

**Comment by kerimoyle** [comment]
This is now unneeded because the bug has been fixed in #520 :) ... but I guess it's worth keeping the tests as they're the only ones which involve multiple MathML blocks.

File: tests/validator/validator.cpp [hunk 1]
```diff
@@ -426,6 +426,50 @@ TEST(Validator, validMath)
     EXPECT_EQ(size_t(0), v->errorCount());
 }
 
+TEST(Validator, validMathInMultipleMathMlBlocks)
```
Comment (agarny) [inline tests/validator/validator.cpp]: Casing, it should be: `validMathInMultipleMathMLBlocks`.

File: tests/validator/validator.cpp [hunk 2]
```diff
@@ -1162,6 +1206,40 @@ TEST(Validator, resetNoResetValue)
     EXPECT_EQ(expectedError, validator->error(0)->description());
 }
 
+TEST(Validator, resetMultipleMathMlInResetValue)
```
Comment (agarny) [inline tests/validator/validator.cpp]: Casing, it should be: `resetMultipleMathMLInResetValue `.

File: tests/validator/validator.cpp [hunk 3]
```diff
@@ -1200,6 +1278,40 @@ TEST(Validator, resetNoTestValue)
     EXPECT_EQ_ERRORS(expectedErrors, validator);
 }
 
+TEST(Validator, resetMultipleMathMlInTestValue)
```
Comment (agarny) [inline tests/validator/validator.cpp]: Casing, it should be: `resetMultipleMathMLInTestValue `.

**Review by agarny — CHANGES_REQUESTED** [review]
I would merge those three tests into one. Especially the last two, they are 99% the same.

File: tests/validator/validator.cpp (hunk 3 above)
Comment (kerimoyle) [reply tests/validator/validator.cpp]: Gah ... in earlier PRs I was told it went the other way  ... :|

**Comment by hsorby** [comment]
> I would merge those three tests into one. Especially the last two, they are 99% the same.

While they are 99% the same I wouldn't merge these into one as they are testing different API calls.

File: tests/validator/validator.cpp (hunk 3 above)
Comment (agarny) [reply tests/validator/validator.cpp]: TBH, I would personally use `resetMultipleMathmlInTestValue`, but I have seen `xxxYyyyMathMLZzzz` in quite a few places.

Comment (kerimoyle) [reply tests/validator/validator.cpp]: I've changed it once ... and you have edit permission ... so you can feel free to change it again if the mood takes you ;)

**Review by nickerso — APPROVED** [review]
I agree that it would be useful to keep these tests as they are not covered elsewhere. Might be useful to start making some functions to do the common setup tasks to address the comment by @agarny , but that can be done as a separate clean up later.

**Review by agarny — APPROVED** [review]

**Pull request merged**